snuba/query/processors/logical/filter_in_select_optimizer.py

```diff
@@ -45,9 +53,14 @@ def process_mql_query(
     ) -> None:
         feat_flag = get_int_config("enable_filter_in_select_optimizer", default=0)
         if feat_flag:
+            metrics.increment("kyles_optimizer_ran")
```

Comment:
you probably want to add something that is actually conditional on the optimizer doing something. This is too high up I think

Reply:
I moved it down to the point where the condition of a query is actually modified